kubeadm/kubeadm.go

```diff
@@ -31,6 +49,7 @@ func New() *Kubeadm {
 	return NewWithCmdRunner(cmd_runner.New())
 }
 
+// TokenCreate execs `kubectl token create` with the appropriate flags added by interpreting the params argument.
```

Comment:
(nit) I think it should say the what and not the how of the method. Users should care about the what and not the how. For example, if you end up changing the implementation to create kubeadm tokens from the library instead of exec from the command line, it does not actually change what this method does from the user perspective.

Reply:
Agree with this sentiment, however, I think at this point this file / method is an actual wrapper for kubeadm and has no other functionality (by design). So the What is the fact that kubectl token create is being execed.

If this method were an interface where the implemention can and will be swapped out then I think this comment applies. If we were to bring in a library form, we wouldn't wrap it here as this is harder to use than it needs to be to be and we would drop the TokenCreateParams struct. Instead, we would either just call it where code is calling this method right now, or we would create a new appropriately named wrapper (tokengen might be a good name? something like that). 

I think this gets back to the on-going theme I have in the comment with @karan that we need an actual token generator that the various machine actuators directly use and then that token generator implementation can use kubeadm for now. I added a bit more information to the issue to fix this: https://github.com/kubernetes-sigs/cluster-api/issues/333.